packages/react/src/components/Button/Button.tsx

```diff
@@ -147,21 +152,31 @@ class Button extends UIComponent<WithAsProp<ButtonProps>, ButtonState> {
         {Box.create(!hasChildren && content, {
           defaultProps: { as: 'span', styles: styles.content },
         })}
-        {!hasChildren && iconPosition === 'after' && this.renderIcon(variables, styles)}
+        {!hasChildren && iconPosition === 'after' && !loading && this.renderIcon(variables, styles)}
```

Comment:
In all cases that I found the Loader was never after the content, that's why I added this condition. We may reconsider this...